src/Internal/GeneratorPipeline.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Amp\Pipeline\Internal;
+
+use Amp\Cancellation;
+use Amp\Pipeline\Pipeline;
+use Amp\Pipeline\PipelineOperator;
+use Revolt\EventLoop;
+
+/**
+ * @template TValue
+ * @template-implements Pipeline<TValue>
+ * @template-implements \IteratorAggregate<int, TValue>
+ */
+final class GeneratorPipeline implements Pipeline, \IteratorAggregate
+{
+    /** @var EmitSource<TValue> */
+    private EmitSource $source;
+
+    public function __construct(\Generator $generator)
+    {
+        $this->source = $source = new EmitSource;
+
+        EventLoop::queue(static function () use ($generator, $source): void {
+            try {
+                foreach ($generator as $value) {
+                    $source->yield($value);
+                }
+
+                $source->complete();
+            } catch (\Throwable $exception) {
+                $source->error($exception);
+            }
+        });
+    }
+
+    /**
+     * @inheritDoc
+     *
+     * @psalm-return TValue|null
+     */
+    public function continue(?Cancellation $cancellation = null): mixed
+    {
+        return $this->source->continue();
```

Comment:
This misses passing `$cancellation`?

Reply:
So it does, seems `AsyncGenerator` does right now too. Copy/pasta. 😅